### ISSUE TITLE
error when unix move-file accross mount points

### ISSUE THREAD

**Issue by jonenst** [factor/factor#1772]
I have separate partitions for / and /home
```
$ mount
/dev/sda1 on / 
/dev/sda5 on /home
```
When running the following
```factor
"/tmp/bar" "/home/jon/bar" move-file
```
I get this exception
```
Unix system call “rename” failed:

Invalid cross-device link (18)

It was called with the following arguments:

"/tmp/bar2"
"/home/jon/bar2"
```

```M\ unix move-file``` should check for EXDEV and default to copy then delete

**Comment by bpollack** [comment]
I actually think we should error here. Moving files is an atomic operation on all file systems that Factor supports, which makes it useful in _tons_ of IPC situations. Keeping that trivially available in Factor makes a lot of sense to me.

That said, we're not being consistent right now across platforms. A quick check shows that we're using `MoveFile` on Win32, which will happily move files (and even directories, though not recursively) across volumes by falling back on `CopyFile`.

So...thoughts? My preferred fix is actually to leave Unix as-is and to change Windows to use `MoveFileEx` without `FILEMOVE_COPY_ALLOWED`, but doing that, or having `move-file` switch to a copy/remove sequence on Unix are almost identical amounts of code, so which we pick is entirely up to what y'all wanna do here.

**Comment by mrjbq7** [comment]
And then a `really-move-file-please` that does the `copy-file` fallback?

**Comment by mrjbq7** [comment]
Consistency is really important, I agree.

**Comment by bpollack** [comment]
@mrjbq7 I think you're kidding, but after reflecting, I think modifying `move-file` to Do The Right Thing and then adding `move-file-atomically` as a publicly documented word that only works on single volumes or whatever. That'd both be least-surprise, _and_ keep the atomic Unix rename prims available. If that sounds good, I'll do that in a bit.

**Comment by mrjbq7** [comment]
Haha, I was kidding!  But I was also saying people might want the other behavior so maybe we can offer that somehow.  Either by ``move-file`` and ``move-file-atomically`` or by ``move-file`` and ``move-file!`` or whatever we name it...

I like the atomic unix primitives:

https://rcrowley.org/2010/01/06/things-unix-can-do-atomically.html

**Comment by bpollack** [comment]
I'll throw a PR up after I'm off this call. Any preference between Windows failing or using distributed transaction semantics when attempting to move files across volumes/the network? For my sanity, I'd prefer to just fail, but whatever. :)

**Comment by bpollack** [comment]
I think the name should be `move-file-atomically`, BTW. `move-file!` doesn't make sense, and `move-file*`, while obeying our naming conventions, is needlessly vague, IMHO. That said, I don't really care much about that.

**Comment by mrjbq7** [comment]
You decide!

**Comment by AlexIljin** [comment]
Maybe use the name `rename-file` for moving file within a filesystem?

**Comment by mrjbq7** [comment]
Fixed by #1775.
